omnigibson/robots/behaviorbot.py

```diff
@@ -0,0 +1,677 @@
+from abc import ABC
+from collections import OrderedDict
+import itertools
+import numpy as np
+import os
+from pxr import Gf
+from scipy.spatial.transform import Rotation as R
+from typing import List, Tuple, Iterable
+
+import omnigibson as og
+from omnigibson.macros import gm
+from omnigibson.robots.locomotion_robot import LocomotionRobot
+from omnigibson.robots.manipulation_robot import ManipulationRobot, GraspingPoint
+from omnigibson.robots.active_camera_robot import ActiveCameraRobot
+from omnigibson.objects.usd_object import USDObject
+import omnigibson.utils.transform_utils as T
+from omnigibson.controllers.controller_base import ControlType
+
+# component suffixes for the 6-DOF arm joint names
+COMPONENT_SUFFIXES = ['x', 'y', 'z', 'rx', 'ry', 'rz']
+
+# Offset between the body and parts
+HEAD_TO_BODY_OFFSET = ([0, 0, -0.4], [0, 0, 0, 1])
+RH_TO_BODY_OFFSET = ([0, 0.15, -0.4], T.euler2quat([0, np.pi, np.pi / 2]))
+LH_TO_BODY_OFFSET = ([0, -0.15, -0.4], T.euler2quat([0, np.pi, -np.pi / 2]))
+
+# Hand parameters
+HAND_GHOST_HAND_APPEAR_THRESHOLD = 0.15
+THUMB_2_POS = [0, -0.02, -0.05]
+THUMB_1_POS = [0, -0.015, -0.02]
+PALM_CENTER_POS = [0, -0.04, 0.01]
+PALM_BASE_POS = [0, 0, 0.015]
+FINGER_TIP_POS = [0, -0.025, -0.055]
+
+# Hand link index constants
+PALM_LINK_NAME = "palm"
+FINGER_MID_LINK_NAMES = ("Tproximal", "Iproximal", "Mproximal", "Rproximal", "Pproximal")
+FINGER_TIP_LINK_NAMES = ("Tmiddle", "Imiddle", "Mmiddle", "Rmiddle", "Pmiddle")
+THUMB_LINK_NAME = "Tmiddle"
+
+
+class Behaviorbot(ManipulationRobot, LocomotionRobot, ActiveCameraRobot):
```

Comment:
is this just the old BehaviorRobot brought back?

Reply:
It's based on my work last summer, which is based on the old BehaviorRobot in iG2.